src/evaluate.jl

```diff
@@ -96,43 +96,42 @@ end
 Evaluate a `HomogeneousPolynomial` polynomial using Horner's rule (hand coded)
 at `vals`.
 """
-function evaluate{T<:Number,S<:Union{Real,Complex}}(a::HomogeneousPolynomial{T},
-    vals::Array{S,1} )
-
-    numVars = get_numvars()
-    @assert length(vals) == numVars
+# function evaluate{T<:Number}(a::HomogeneousPolynomial{T}, vals::Array{T,1} )
+function evaluate{T<:Number,S<:NumberNotSeriesN}(a::HomogeneousPolynomial{T},
+        vals::Array{S,1} )
+    @assert length(vals) == get_numvars()
     R = promote_type(T,S)
+    numVars = get_numvars()
     suma = convert(TaylorN{R}, a)
 
-    for nv = 1:numVars
+    @inbounds for nv = 1:numVars
         suma = horner(suma, (nv, vals[nv]))
     end
 
     return suma.coeffs[1].coeffs[1]
```

Comment:
Maybe. The way it is done here is that a `TaylorN` polynomial is evaluated substituting each variable, one by one; maybe it's not the best way of doing it...